seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-jdbc-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/JdbcSqlserverIT.java

```diff
@@ -0,0 +1,148 @@
+                "  [ximage] image  NULL\n" +
+                ")";
+            String sinkSql = "CREATE TABLE [sink] (\n" +
+                "  [ids] bigint  NOT NULL,\n" +
+                "  [name] text COLLATE Chinese_PRC_CI_AS  NULL,\n" +
+                "  [sfzh] varchar(255) COLLATE Chinese_PRC_CI_AS  NULL,\n" +
+                "  [sort] int  NULL,\n" +
+                "  [dz] nvarchar(255) COLLATE Chinese_PRC_CI_AS  NULL,\n" +
+                "  [xchar] char(255) COLLATE Chinese_PRC_CI_AS  NULL,\n" +
+                "  [xdecimal] decimal(18)  NULL,\n" +
+                "  [xfloat] float(53)  NULL,\n" +
+                "  [xnumeric] numeric(18)  NULL,\n" +
+                "  [xsmall] smallint  NULL,\n" +
+                "  [xbit] bit  NULL,\n" +
+                "  [rq] datetime DEFAULT NULL NULL,\n" +
+                "  [xrq] smalldatetime  NULL,\n" +
+                "  [xreal] real  NULL,\n" +
+                "  [ximage] image  NULL\n" +
+                ")";
+            statement.execute(sourceSql);
+            statement.execute(sinkSql);
+        } catch (SQLException e) {
+            throw new RuntimeException("Initializing Sqlserver table failed!", e);
+        }
+    }
+
+    @SuppressWarnings("checkstyle:RegexpSingleline")
+    private void batchInsertData() {
+        try (Connection connection = DriverManager.getConnection(mssqlServerContainer.getJdbcUrl(), mssqlServerContainer.getUsername(), mssqlServerContainer.getPassword())) {
+            String sql =
+                "INSERT INTO [source] ([ids], [name], [sfzh], [sort], [dz], [xchar], [xdecimal], [xfloat], [xnumeric], [xsmall], [xbit], [rq], [xrq], [xreal], [ximage]) " +
+                    "VALUES (1504057, '张三', '3ee98c990e2011eda8fd00ff27b3340d', 1, N'3232', 'qwq', 1, 19.1, 2, 1, '0', '2022-07-26 11:58:46.000', '2022-07-26 13:49:00', 2, 0x)";
+            Statement statement = connection.createStatement();
+            statement.execute(sql);
+        } catch (SQLException e) {
+            throw new RuntimeException("Batch insert data failed!", e);
+        }
+    }
+
+    @Test
+    public void tesSqlserverSourceAndSink() throws SQLException, IOException, InterruptedException {
+        Container.ExecResult execResult = executeSeaTunnelFlinkJob("/jdbc/jdbc_sqlserver_source_to_sink.conf");
+        Assertions.assertEquals(0, execResult.getExitCode());
+        // query result
+        String sql = "select * from sink";
+        try (Connection connection = DriverManager.getConnection(mssqlServerContainer.getJdbcUrl(), mssqlServerContainer.getUsername(), mssqlServerContainer.getPassword())) {
+            Statement statement = connection.createStatement();
+            ResultSet resultSet = statement.executeQuery(sql);
+            List<String> result = Lists.newArrayList();
+            while (resultSet.next()) {
+                result.add(resultSet.getString("ids"));
+            }
+            Assertions.assertFalse(result.isEmpty());
```

Comment:
Check whether the data in the source table and sink table are consistent, just checking the amount of data is not accurate enough